### PR TITLE
Add support for Ernie NLP model to the Slim QAT

### PR DESCRIPTION
**Pull request by wojtuss**
This PR adds support for the Ernie NLP model in QAT2.0.
A documentation update explaining how to run accuracy and performance benchmarks on this model is submitted in a separate PR: https://github.com/PaddlePaddle/Paddle/pull/22519.
For the performance benchmark to work, the PR https://github.com/PaddlePaddle/benchmark/pull/311 needs to be merged.

### CODE REVIEW

File: python/paddle/fluid/contrib/slim/quantization/quantization_mkldnn_pass.py [hunk 1]
```diff
@@ -371,34 +387,94 @@ def _gather_scales(self, graph):
...
+
+        while len(waiting_for_scale) != 0:
+            waiting_for_scale = _update_scales(graph)
+
+        return graph
+
+    def _set_dummy_fc_out_scales(self, graph):
+        '''
+        For the output tensors of FC that do not have an assigned scale,
+        assign a dummy scale (same scale as input), so tha the quantize pass
```
Comment (Sand3r-) [inline python/paddle/fluid/contrib/slim/quantization/quantization_mkldnn_pass.py]: tha -> that.

**Comment by bingyanghuang** [comment]
这个PR做了3件事：

1. 在cmake文件中创建QAT目录，用于分别放置int8v2的模型和QAT相关的模型，之前一律放到int8v2中不是很干净
2. 更改slim中mkldnn QAT quantization的名字，使其更简洁更短
3. 新增将ERNIE fake QAT模型转换为mkldnn INT8模型的相关函数

@wzzju @juncaipeng 希望这样可以帮助你们快速的review这个PR。

**Comment by luotao1** [comment]
Please add `test=develop` to trigger the CI

File: python/paddle/fluid/contrib/slim/tests/qat_int8_image_classification_comparison.py [hunk 2]
```diff
@@ -68,17 +64,25 @@ def parse_args():
...
 
-class TestQatInt8Comparison(unittest.TestCase):
+class QatInt8ImageClassificationComparisonTest(unittest.TestCase):
     """
-    Test for accuracy comparison of QAT FP32 and INT8 inference.
+    Test for accuracy comparison of QAT FP32 and INT8 Image Classification inference.
     """
 
+    #  def __init__(self, debug):
+    #  self._debug = debug
```
Comment (Sand3r-) [inline python/paddle/fluid/contrib/slim/tests/qat_int8_image_classification_comparison.py]: These commented lines should be removed

**Review by wzzju — DISMISSED** [review]
LGTM.

File: python/paddle/fluid/contrib/slim/quantization/quantization_mkldnn_pass.py (hunk 1 above)
Comment (Sand3r-) [reply python/paddle/fluid/contrib/slim/quantization/quantization_mkldnn_pass.py]: Done.

File: python/paddle/fluid/contrib/slim/tests/qat_int8_image_classification_comparison.py (hunk 2 above)
Comment (Sand3r-) [reply python/paddle/fluid/contrib/slim/tests/qat_int8_image_classification_comparison.py]: Done.

**Review by wozna — DISMISSED** [review]
Good work!  LGTM

**Review by wzzju — DISMISSED** [review]
LGTM.

**Review by juncaipeng — COMMENTED** [review]
LGTM  for the unittest

**Review by luotao1 — APPROVED** [review]

**Pull request merged**